patches/tModLoader/Terraria/ModLoader/Core/AssemblyManager.cs

```diff
@@ -202,7 +202,8 @@ private static Mod Instantiate(ModLoadContext mod)
 			m.Side = mod.properties.side;
 			m.DisplayName = mod.properties.displayName;
 			m.TModLoaderVersion = mod.properties.buildVersion;
-			m.TranslationForMods = mod.properties.translationMod ? mod.properties.RefNames(true).ToList() : null; 
+			m.TranslationForMods = mod.properties.translationMod ? mod.properties.RefNames(true).ToList() : null;
+			m.content = ContentCache.GetOrCreate(m);
```

Comment:
No need for this to be a `GetOrCreate`, since `m` is always a new `Mod`. Thus `ContentCache._sourcesByMod` will also be unnecessary.

This should be initialized in the `Mod` constructor.